paddle/fluid/memory/jemalloc_test.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <jemalloc/jemalloc.h>
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+
+template <typename MallocFunc, typename... Args>
+std::unique_ptr<void, void (*)(void *)> je_malloc_func(MallocFunc &&func,
+                                                       Args... args) {
+  std::unique_ptr<void, void (*)(void *)> ret(func(std::forward<Args>(args)...),
+                                              je_free);
+  return ret;
+}
+
+TEST(jemalloc, jemalloc_not_overwrite_std) {
+  ASSERT_TRUE(malloc != je_malloc);
+  ASSERT_TRUE(calloc != je_calloc);
+  ASSERT_TRUE(aligned_alloc != je_aligned_alloc);
+  ASSERT_TRUE(realloc != je_realloc);
+
+#ifdef _WIN32
+  ASSERT_TRUE(_aligned_malloc != je_aligned_alloc);
+#else
+  ASSERT_TRUE(posix_memalign != je_posix_memalign);
+#endif
+
+  ASSERT_TRUE(free != je_free);
+}
+
+size_t n = (1 << 20);
+size_t kAlignment = 64;
+
+TEST(jemalloc, je_malloc) {
+  auto x = je_malloc_func(je_malloc, n);
+  ASSERT_NE(x, nullptr);
+}
+
+TEST(jemalloc, je_calloc) {
+  using DataType = float;
+  auto x = je_malloc_func(je_calloc, n, sizeof(DataType));
+  ASSERT_NE(x, nullptr);
+  bool is_all_zero = std::all_of(
+      static_cast<DataType *>(x.get()), static_cast<DataType *>(x.get()) + n,
+      [](const DataType &val) { return val == static_cast<DataType>(0); });
+  ASSERT_TRUE(is_all_zero);
```

Comment:
it will erase as zero in all platform? Not sure about this test case.

Reply:
`calloc` does guarantee it.